Document auth rule backup support and add temporal warnings to ABAC docs (#2974)
neo4j/docs-operations: modules/ROOT/pages/authentication-authorization/attribute-based-access-control.adoc

+
[NOTE]
====

> The functions, `date`, `datetime`, `localdatetime`, `localtime`, and `time` are only supported when you use an input argument to specify the instance. This is because auth rules get evaluated at the instant that the transaction begins.

Reviewer: ```suggestion
The functions `date`, `datetime`, `localdatetime`, `localtime`, and `time` are only supported when you use an input argument to specify the instant.
This is because auth rules are evaluated at the moment the transaction begins.
```

Reviewer: What does instance mean in this case?

Author: oh, it should be instant instead of instance. But I like your suggestion, so it would be:

```
The functions `date`, `datetime`, `localdatetime`, `localtime`, and `time` are only supported when you use an input argument to specify the instant.
This is because auth rules are evaluated at the moment the transaction begins.

> As far as auth rules are concerned, there is no concept of statement time or real time. To get the current temporal instant, use any of the `*.transaction()` functions.

Reviewer: ```suggestion
As far as auth rules are concerned, there is no concept of statement time or real time. 
To get the current temporal instant type, use any of the `*.transaction()` functions.
```

====
+
[NOTE]
====
Transaction time functions such as `datetime.transaction()` always reflect the server's clock time.

> In a cluster, it is important to keep the system clocks synchronized across all cluster nodes to ensure consistent auth rule evaluation.

Reviewer: ```suggestion
In a cluster, it is important to keep the system clocks synchronized across all cluster members to ensure consistent auth rule evaluation.
```

====
+

> [WARNING]

Reviewer: Too many notes and warnings. Can we group them somehow?

Reviewer: @HannesSandberg, how about grouping them in one note and one warning instead:
```
[NOTE]
====
The functions `date`, `datetime`, `localdatetime`, `localtime`, and `time` are only supported when you use an input argument to specify the instant.
This is because auth rules are evaluated at the moment the transaction begins.
As far as auth rules are concerned, there is no concept of statement time or real time. 
To get the current temporal instant type, use any of the `*.transaction()` functions.

Transaction time functions such as `datetime.transaction()` always reflect the server's clock time.
In a cluster, it is important to keep the system clocks synchronized across all cluster members to ensure consistent auth rule evaluation.
====

[WARNING]
====
* Avoid using `localtime` and `localdatetime` without specifying a timezone, as these functions rely on the server's default timezone.

* Avoid mixing time values from authentication token claims (via `abac.oidc.user_attribute()`) with server time functions such as `datetime.transaction()` or `time()`.
Token timestamps are fixed at the time of issuance and may use different formats or time zones, leading to unexpected comparison results.
====
```

Author: Sounds like a good idea!

====
Avoid using `localtime` and `localdatetime` without specifying a timezone, as these functions rely on the server's default timezone.
====
+
[WARNING]
====
Avoid mixing time values from authentication token claims (via `abac.oidc.user_attribute()`) with server time functions such as `datetime.transaction()` or `time()`.

> Token timestamps are fixed at the time of issuance and may use different formats or timezones, which can lead to unexpected comparison results.

Reviewer: ```suggestion
Token timestamps are fixed at the time of issuance and may use different formats or time zones, which can lead to unexpected comparison results.
```

> The functions, `date`, `datetime`, `localdatetime`, `localtime`, and `time` are only supported when you use an input argument.

Reviewer 2: yes.

        AND datetime.transaction().hour >= 19 AND datetime.transaction().hour < 20;
----

The following example uses zoned time anchored to UTC to avoid dependency on the server's default timezone, granting access to users in the `EMEA` region during European working hours (06:00–18:00 UTC):

[source, cypher25, role="noheader"]
----
CREATE AUTH RULE emea_working_hours
    SET CONDITION abac.oidc.user_attribute('region') = 'EMEA'
        AND time({timezone: 'UTC'}).hour >= 6 AND time({timezone: 'UTC'}).hour < 18;
----

[WARNING]
====
Avoid mixing time values from authentication token claims (via `abac.oidc.user_attribute()`) with server time functions such as `datetime.transaction()` or `time()`.
Token timestamps are fixed at the time of issuance and may use different formats or timezones, which can lead to unexpected comparison results.

> ====

Reviewer 2: yes that makes more sense. One minor change:

in this paragraph: 

> As far as auth rules are concerned, there is no concept of statement time or real time. To get a temporal instance without specifying the instance, use any of the *.transaction() functions.

_instance_ should be _instant_  I think.

> ====

Author: I think you are right, I also change the wording a little bit. to not mention instant twice

> ====

Reviewer 2: nice
